### PR TITLE
Add support for multiple article pictures

### PR DESCRIPTION
**Pull request by lucaslavandeira**
Extensively modifies the Edit Article activity and adds new activities
and layouts to support an article having more than one picture.

The logic for saving the pictures is modified: the backend will now
store the direct image URLs, and they will be served directly instead of
reading from the Firebase storage.

The UI for multiple images is implemented with a Slider library. The
selection is handled from another activity with a Grid View, that uses
an Image adapter to dynamically add and remove images, and the final
state is passed on as an array of URLs back to the main edit article
activity.

The Article model is now always initialzed with null values as soon as
the edit article activity is created, and values are read from the UI
and passed on to setters when the user clicks save. This allows us to
store pictures on the model directly, whenever the user decides to add
some, instead of keeping a temporary array and passing it onto the model
when we need to hit the backend. Some Text Watcher logic to
enable/disable the save button was modified to allow this change.

With this change, the FirebaseImageManager util class becomes a little
obsolete, since we store the image URLs directly, we don't need to go
through the storage. Maybe the update account logic should follow the
same pattern of storing URLs to fully deprecate the module.

### CODE REVIEW

File: app/src/main/java/ar/uba/fi/mercadolibre/activity/EditImageActivity.java [hunk 1]
```diff
@@ -0,0 +1,167 @@
...
+        if (article == null) {
+            return;
+        }
+        initAdapter();
+    }
+
+    @Override
+    public boolean onCreateOptionsMenu(Menu menu) {
+        getMenuInflater().inflate(R.menu.menu_edit_images, menu);
+        return true;
```
Comment (dylanalvarez) [inline app/src/main/java/ar/uba/fi/mercadolibre/activity/EditImageActivity.java]: yo a estos true le pondria nombre, algo como

bool whateverThatTrueImplies = true
return true

File: app/src/main/java/ar/uba/fi/mercadolibre/activity/EditImageActivity.java [hunk 2]
```diff
@@ -0,0 +1,167 @@
...
+import com.squareup.picasso.Callback;
+
+import java.util.ArrayList;
+
+import ar.uba.fi.mercadolibre.R;
+import ar.uba.fi.mercadolibre.adapter.ImageAdapter;
+import ar.uba.fi.mercadolibre.model.Article;
+import ar.uba.fi.mercadolibre.utils.FirebaseImageManager;
+
+public class EditImageActivity extends BaseActivity {
```
Comment (dylanalvarez) [inline app/src/main/java/ar/uba/fi/mercadolibre/activity/EditImageActivity.java]: Este nombre puesto suelto en activity me da la idea de que EditImageActivity no esta acoplado a un modelo en particular, le pondria un nombre que tenga Article en algun lado, o lo meteria en un package activity.article

Por suerte igual creo que no necesitamos un componente como este en otro lado, asi que no creo que sea necesario eventualmente desacoplar la logica mas general de la especifica de Article

File: app/src/main/java/ar/uba/fi/mercadolibre/activity/EditArticleActivity.java [hunk 3]
```diff
@@ -41,10 +39,10 @@
 
 public class EditArticleActivity extends BaseActivity {
     private FusedLocationProviderClient mFusedLocationClient;
+    private Article article = null;
 
-    private static final int GET_FROM_GALLERY = 3;
     private static final int REQUEST_COARSE_LOCATION = 1;
-    Uri newImageUri = null;
+    private static final int EDIT_IMAGES = 2;
```
[the remark this answers is not in the document]
Comment (lucaslavandeira) [reply app/src/main/java/ar/uba/fi/mercadolibre/activity/EditArticleActivity.java]: Los números son IDs arbitrarios que vos como programador setéas, le decís a android "che empezame esta nueva activity con código X", entonces cuando esa actividad termina, se ejecuta el `onActivityResult` con el parámetro `requestCode` igual al ID que le pasaste. Es la manera de diferenciar de dónde viene el resultado que triggereó ese evento

File: app/src/main/java/ar/uba/fi/mercadolibre/activity/EditImageActivity.java (hunk 1 above)
Comment (lucaslavandeira) [reply app/src/main/java/ar/uba/fi/mercadolibre/activity/EditImageActivity.java]: uh pero son parte del framework

**Pull request merged**